pyvista/core/filters/data_set.py

```diff
@@ -5088,7 +5176,12 @@ def transform(
         accepted.
 
         >>> transform_matrix = np.array(
-        ...     [[1, 0, 0, 50], [0, 1, 0, 100], [0, 0, 1, 200], [0, 0, 0, 1]]
+        ...     [
+        ...         [1, 0, 0, 50],
+        ...         [0, 1, 0, 100],
+        ...         [0, 0, 1, 200],
+        ...         [0, 0, 0, 1],
+        ...     ]
```

Comment:
Local testing shows that it doesn't matter. I'm guessing it's internally converted to float because I get the same results regardless of the input datatype.